articles/hapi-vs-koa-vs-express/index.md

```diff
@@ -0,0 +1,159 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/hapi-vs-koa-vs-express/
+title: Hapi vs Koa vs Express
+description: In this article we do a comparative analysis of three Node.js frameworks - we explore their advantages, disadvantages, and work an example to run a server in each.
+author: peter-kayere
+date: 2020-09-21T00:00:00-08:00
+topics: [Node.js]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/hapi-vs-koa-vs-express/hero.jpg
+    alt: koa koa.js Node.js express hapi framework
+---
+Node.js has various frameworks each of which is catered to the demands of users and developers. This article will give a comparison between the three major Node.js frameworks i.e. [express](https://expressjs.com), [hapi](https://hapi.dev) and [koa](https://koajs.com). In the comparison we will look at the difference between the three frameworks, their histories, advantages and disadvantages, popularity, support and performance.
+<!--more-->
+For more background on the different types of frameworks in Node.js, check out the following articles to get a more in depth look at Hapi, Koa, and Node.js before we begin:
```

Comment:
background on the **frameworks** would more succinct as well as removing at Hapu, Koa, and Node.js.